guidelines/design-guidelines.md

```diff
@@ -0,0 +1,34 @@
+## Configuration API's
+
+All configuration API's should be code first since that allows us to best evolve them while guiding users with obsolete messages.
+
+Configuration API's should be non fluent and non delegate based. Not returning `this` means that they are not chainable.
```

Comment:
Suggest:

>Configuration API's should be ~non fluent~ *non-fluent* and ~non delegate~ *non-delegate* based. ~Not returning~ *They should not return* `this`, ~means that they are not chainable~ *to avoid being chain-able*.